src/io/flutter/coverage/FileCoverageInfo.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.coverage;
+
+import com.intellij.openapi.util.Pair;
+import com.intellij.openapi.vfs.VirtualFile;
+import gnu.trove.TIntHashSet;
+
+class FileCoverageInfo {
+  private final VirtualFile file;
+
+  private final TIntHashSet coveredLines = new TIntHashSet();
+  private final TIntHashSet uncoveredLines = new TIntHashSet();
+
+  public FileCoverageInfo(VirtualFile file) {
+    this.file = file;
+  }
+
+  public VirtualFile getFile() {
+    return file;
+  }
+
+  public void addCovered(Pair<Integer, Integer> pos) {
+    if (pos == null) {
+      return;
+    }
+
+    final int line = pos.first;
+    coveredLines.add(line);
+    uncoveredLines.remove(line);
+  }
+
+  public void addUncovered(int line) {
+    if (!coveredLines.contains(line)) {
+      uncoveredLines.add(line);
+    }
+  }
+
+  public void addUncovered(Pair<Integer, Integer> pos) {
+    if (pos == null) {
+      return;
+    }
+    final int line = pos.first;
+    if (!coveredLines.contains(line)) {
```

Comment:
Maybe replace with a call to `addUncovered(line)` or `addUncovered(pos.first)`?

Reply:
I'd like to keep this as is - I think this method (and the other `addUncovered()` method, and `addCovered`) can have knowledge of manipulating the `coveredLines` and `uncoveredLines` data structures.